repository/src/main/java/org/apache/atlas/discovery/EntityLineageService.java

```diff
@@ -203,20 +220,24 @@ public AtlasLineageListInfo getLineageListInfoOnDemand(String guid, LineageListR
         return ret;
     }
 
-    private boolean validateEntityTypeAndCheckIfDataSet(String guid) throws AtlasBaseException {
+    private HashMap<String, Boolean> validateAndGetEntityTypeMap(String guid) throws AtlasBaseException {
```

Comment:
The return type of this method is HashMap<String, Boolean>, all I can see this map is used to get a boolean based on following if condition
```
        boolean isConnecterVertex =  lineageType.equals(PRODUCT_ASSET_LINEAGE)
                ? !dataTypeMap.get(IS_DATA_PRODUCT)
                : !dataTypeMap.get(IS_DATASET);
```


Can you check if you can directly return the required boolean flag from this method?